packages/roosterjs-editor-dom/lib/edit/adjustInsertPosition.ts

```diff
@@ -235,9 +242,36 @@ function adjustInsertPositionForMoveCursorOutOfALink(
     return position;
 }
 
+/**
+ * Adjust the cursor position when a image is inserted to prevent it be inserted into another element.
+ */
+function adjustInsertPositionForImages(
+    root: HTMLElement,
+    nodeToInsert: Node,
+    position: NodePosition,
+    range: Range
+): NodePosition {
+    const isImage = nodeToInsert.nodeName === 'IMG' ? true : false;
+    if (range && range.collapsed && isImage) {
+        const searcher = new PositionContentSearcher(root, Position.getStart(range));
+        const inlineElementBefore = searcher.getInlineElementBefore();
+        const inlineElementAfter = searcher.getInlineElementAfter();
+        const adjustedPosition = (inlineElement: InlineElement, positionType: PositionType) =>
```

Comment:
I'm not a fan of this kind of inline function. Make it a regular function if we can.